src/praisonai-deploy/praisonai_deploy/api.py

```
@@ -149,23 +150,29 @@ def start_api_server(
-        # Install flask and flask-cors if needed
+        # Install flask and flask-cors if needed. Use the same interpreter that
+        # will run the server so packages land where the child process expects.
         try:
             subprocess.run(
-                ['pip', 'install', '-q', 'flask', 'flask-cors'],
+                [sys.executable, '-m', 'pip', 'install', '-q', 'flask', 'flask-cors'],
                 check=False,
                 capture_output=True
             )
         except Exception:
             pass
```

Ruff (0.16.0)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)